src/logging.cpp

```diff
@@ -273,17 +318,45 @@ void BCLog::Logger::LogPrintStr(const std::string& str, const std::string& loggi
     if (m_print_to_file) {
         assert(m_fileout != nullptr);
 
-        // reopen the log file, if requested
-        if (m_reopen_file) {
-            m_reopen_file = false;
+        auto reopen_file = [&]() {
             FILE* new_fileout = fsbridge::fopen(m_file_path, "a");
             if (new_fileout) {
                 setbuf(new_fileout, nullptr); // unbuffered
                 fclose(m_fileout);
                 m_fileout = new_fileout;
+                m_file_size = 0;
             }
```

Comment:
Not sure this should be a lambda. The advantage is that it's pretty small and it's declared close to its two call sites, but this is a slightly unusual practice; probably better as just a normal method in `BCLog::Logger`. Unless there are objections, I'll make that change in the next commit.